entity_recognition/entity_recognition_training.ipynb

```diff
@@ -0,0 +1,871 @@
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of sentences: 20718\n",
+      "Eu rejects German call to boycott British lamb.\n"
+     ]
+    }
+   ],
+   "source": [
+    "print('Number of sentences: %d' % len(sentences)) \n",
+    "print(sentences[0]) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0fb1b2b0",
+   "metadata": {},
+   "source": [
+    "## 3. Train Models using Cross-Validation \n",
+    "\n",
+    "To identify potential label errors in the training dataset, we compute the out-of-sample predicted probabilities using cross-validation. We first partition the dataset into k-folds: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "797df997",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "'folds/' already exists, skipping...\n"
+     ]
+    }
+   ],
+   "source": [
+    "lines = [[]] \n",
+    "for filepath in filepaths: \n",
+    "    for line in open(filepath) : \n",
+    "        if len(line) == 0 or line.startswith('-DOCSTART') or line[0] == '\\n':\n",
+    "            if len(lines[-1]) > 0: \n",
+    "                lines.append([]) \n",
+    "        else: \n",
+    "            lines[-1].append(line) \n",
+    "        \n",
+    "lines = lines[:-1] \n",
+    "lines = [line for m, line in zip(mask, lines) if m] \n",
+    "\n",
+    "k = 5 \n",
+    "indices = create_folds(lines, k=k) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "304653aa",
+   "metadata": {},
+   "source": [
+    "We train one model for each fold's training/testing pair: \n",
+    "\n",
+    "- Warning! The following code will take a long time to execute, and is recommended to run on GPU, otherwise it will take forever. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "6c877f7e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Model 0 already exists, skipping...\n",
+      "Model 1 already exists, skipping...\n",
+      "Model 2 already exists, skipping...\n",
+      "Model 3 already exists, skipping...\n",
+      "Model 4 already exists, skipping...\n"
+     ]
+    }
+   ],
+   "source": [
+    "for i in range(k): \n",
+    "    if os.path.exists('folds/fold%d/model/' % i): \n",
+    "        print('Model %d already exists, skipping...' % i) \n",
+    "    else: \n",
+    "        print('Training model on fold %d (out of %d) of cross-validation...' % (i, k)) \n",
+    "        os.system(\n",
+    "            \"python3 run_ner.py --data_dir=folds/fold%d --bert_model=bert-base-cased \" \\\n",
+    "            \"--task_name=ner --output_dir=folds/fold%d/model --max_seq_length=256 \" \\\n",
+    "            \"--do_train --num_train_epochs 10 --warmup_proportion=0.1\" % (i, i)) \n",
+    "        print('Model %d saved' % i) \n",
+    "        # TODO: make this a function in run_ner.py? "
```

Comment:
Yes please if you're able to!  It's quite weird to call a command line Python from a jupyter notebook when we could just call the python functions directly